torch/_dynamo/variables/functions.py

```diff
@@ -308,7 +308,13 @@ def call_function(
         # since we ensure `forward` of allowed modules can be traced by AOT safely.
         # Note this is not only for allowed modules, as user customized modules can extend from
         # allowed modules but using parent's `forward` method, which is also covered by this branch.
-        if isinstance(self.obj, variables.NNModuleVariable):
+
+        # If we are tracing the higher order op, we want Dynamo to step inside
+        # the module call so that Dynamo can see the underlying parameters and
+        # buffers and raise them as inputs to the graph.
+        if tx.output.is_root_tracer() and isinstance(
```

Comment:
This effectively delegates it to the super() for a non root tracer, right?

Reply:
Correct, which in turn allows Dynamo to trace the innards of `forward` method.

Reply:
Fair enough. Let me add comments to say how `super` means inlining.